### DISCUSSION TITLE
jest types are not recognized when in subfolder

### DISCUSSION THREAD

**Discussion by maxkarkowski** [orta/vscode-jest#687] [Q&A] [answered]
### Environment

1. `node -v`: 10.15.3
1. `npm -v`: 6.13.4
1. `npm ls jest`: 25.1.0
1. your vscode-jest settings if customized:
   "jest.pathToJest": "frontend/node_modules/.bin/jest",
  "jest.pathToConfig": "frontend/jest.config.js",

1. Operating system: MacOS Catalina

### Prerequisite
- are you able to run jest test from command line? yes
- how do you run your tests from command line? yarn test

### Relevant Debug Info

so i have a project splitted up in frontend and backend folder
the frontend is a typescript with parcel to compile it to js and jest for testing. nothing fancy.
the node_modules are also in the frontend folder

The Jest Runner works fine.
but when i open the test.file i get this error "Cannot find name 'test'. Do you need to install type definitions for a test runner? Try `npm i @types/jest` or `npm i @types/mocha`.ts(2582)" when i hover over "test" or "expect"

i installed the types. the file is colored red in the filetree.

i create a example repo (https://github.com/maxkarkowski/jest-vscode-test)
with the vscode settings folder visible in it

### Expected Behavior

vscode should find the types even if they are in a subfodler

### Actual Behavior

test files are marked 

---

**Comment by philipfeldmann** [comment]
Facing the same issue. Unit tests itself are working but intellisense isn't.

Any suggestions?

**Comment by ajschmidt8** [comment]
also experiencing this with a similar setup using https://github.com/kulshekhar/ts-jest

**Comment by hisuwh** [comment]
We're seeing the same problem.  Did anyone find a solution?

**Comment by connectdotz** [comment, marked as answer]
looks like the root problem for @maxkarkowski repo is related to the monorepo setup. I can see a few possible solutions:
1. convert the project to a true [multiroot project](https://code.visualstudio.com/docs/editor/multi-root-workspaces), the cleanest solution IMHO.
2. open the "frontend" as the project root instead of the parent directory
3. remove the "src/**/*.test.ts" from `exclude` option in `frontend/tsconifg.json`

this discussion might also help: #686

**Reply by hisuwh** [reply]
`2. remove the "src/**/*.test.ts" from exclude option in frontend/tsconifg.json`

This was the solution that worked in my case.

**Comment by connectdotz** [comment]
Since this is not related to the extension, I will convert it to a discussion instead.
